sync/src/block/extension.rs

```diff
@@ -1044,10 +1116,15 @@ impl Extension {
 
     fn on_chunk_response(&mut self, from: &NodeId, roots: &[H256], chunks: &[Vec<u8>]) {
         let (block, restore) = match self.state {
-            State::SnapshotChunk {
+            State::SnapshotTopChunk {
                 block,
```

Comment:
It's not related to this commit, but the current code copies a block hash unnecessary. You can fix it with `ref block`.

Reply:
Ah, this PR contains this change too. I saw https://github.com/CodeChain-io/codechain/pull/1929/commits/5a6853c59cb86060d41cd92b6649e0bf6ad0eb9b first, so I thought it was not part of this PR.

Reply:
Ignore these comments, I found that `send_chunk_request` mutably borrows `self`.